Cabal/Distribution/Simple/Build.hs

```diff
@@ -216,7 +216,13 @@ buildComponent verbosity numJobs pkg_descr lbi suffixes
     setupMessage' verbosity "Building" (packageId pkg_descr)
       (componentLocalName clbi) (maybeComponentInstantiatedWith clbi)
     let libbi = libBuildInfo lib
-        lib' = lib { libBuildInfo = addExtraCxxSources (addExtraCSources libbi extras) extras }
```

Comment:
@23Skidoo based on your observation, wasn't this line already broken? i.e. the same FilePaths being added to cxx *and* c sources?

Reply:
Yes, looks like it.